packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -156,13 +176,21 @@ export const ProjectMigratorMWV3: Middleware = async (ctx: CoreHookContext, next
       return false;
     }
 
-    const skipUserConfirm = getParameterFromCxt(ctx, "skipUserConfirm");
-    if (!skipUserConfirm && !(await askUserConfirm(ctx, versionForMigration))) {
-      return;
+    const isRunMigration = await showNotification(ctx, versionForMigration);
+    if (isRunMigration) {
+      const isNonmodalMessage = getParameterFromCxt(ctx, Parameters.isNonmodalMessage);
+      if (isNonmodalMessage) {
+        const versionForMigration = await checkVersionForMigration(ctx);
```

Comment:
why we need to check version again here?

Reply:
nonmodal mode will show the upgrade notification. 
If user click upgrade button in notification after the project is v3, it will trigger migration error.
Double check to exclude this case.